src/filesystem.rs

```diff
@@ -268,6 +270,22 @@ impl ShortFileName {
     const FILENAME_BASE_MAX_LEN: usize = 8;
     const FILENAME_MAX_LEN: usize = 11;
 
+    /// Get base name (name without extension) of file name
+    pub fn base_name(&self) -> &[u8] {
+        let mut first_space: usize = 0;
```

Comment:
Default to 8 here, as a file name may not have any spaces. Worth adding some unit tests for this to catch the edge cases!

Reply:
*scroll*

Lol, there they are ;)